packages/core/src/filters/FilterSystem.ts

```diff
@@ -170,17 +173,28 @@ export class FilterSystem extends System
         state.legacy = legacy;
 
         state.target = target;
-
         state.sourceFrame.copyFrom(target.filterArea || target.getBounds(true));
 
         state.sourceFrame.pad(padding);
+
         if (autoFit)
         {
-            state.sourceFrame.fit(this.renderer.renderTexture.sourceFrame);
+            const sourceFrameProjected = this.renderer.renderTexture.sourceFrame.clone();
+
+            // Project source frame into world space (if projection is applied)
+            this.transformAABB(renderer.projection?.transform?.clone().invert() || Matrix.IDENTITY, sourceFrameProjected);
+
+            state.sourceFrame.fit(sourceFrameProjected);
         }
 
-        // round to whole number based on resolution
-        state.sourceFrame.ceil(resolution);
+        // Round sourceFrame in screen space based on render-texture.
+        this.roundFrame(
+            state.sourceFrame,
+            renderer.renderTexture.current ? renderer.renderTexture.current.resolution : renderer.resolution,
```

Comment:
`renderer.renderTexture` is used a lot! could create a local var?

Reply:
that's a system,. and i really dont want to name `renderTexture` there, so it wont save much

Reply:
So call the local var `renderTextureSystem` doesn't matter if it's verbose, will get minified to a single character anyways

Reply:
Done 👍